### PR TITLE
Add Echo test group

### PR DESCRIPTION
**Pull request by jdlrobson**
- Uses 3 accounts for 3 types of users
- Requires update to Docker image, please reset database to use.
Accounts have been set up with 0, 1 and 100+ notifications.
- Smoke tests covering both Vector and Minerva skins.
- Due to legacy Vector coverage, special affordances are made to handle
the dancing tabs problem and Echo bugs.

This will help with testing of https://phabricator.wikimedia.org/T301263 and later https://phabricator.wikimedia.org/T257143

### CODE REVIEW

**Comment by nicholasray** [comment]
Thanks for the PR! 

I uploaded the db file (database_2022-05-26_13-03-37-0700(PDT).tar.gz) you sent me to netlify so as part of this PR, the database seed script can be updated with the name of your database tar file:

https://github.com/nicholasray/pixel/blob/c6904f401ab6b75b51305e60c4ffe9a0c49f87f5/src/seedDb.sh#L3 

It can be updated to:

```
curl "https://pixeltool.netlify.app/database_2022-05-26_13-03-37-0700(PDT).tar.gz" -o var/lib/mysql/database.tar.gz
```

**Comment by nicholasray** [comment]
For some reason, the `SCENARIO > Talk:Test (#vector)` scenario is hanging (and never completing). You should be able to replicate it with `./pixel.js reference` (please let me know if you can't)

File: configEcho.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+ * @param {Object} additionalConfig
+ * @return {Array} of scenarios.
+ */
+const makeScenarios = ( hashtags, queryString, additionalConfig = {} ) => {
+	return ( test ) => {
+		const hashtagString = hashtags.concat( [ '#logged-in', '#echo' ] ).concat(
+			test.hashtags
+		).join( ' ' );
+		return Object.assign( {
+			selectors: [ 'viewport' ]
```
Comment (nicholasray) [inline configEcho.js:58]: I'm seeing the following with the `viewport` selector`:
before

Changing the selector to `.client-js` (on the html element) leads to:

MediaWiki_Echo_smoke_test_with_0_notifications_vector-2022_logged-in_echo_echo-drawer_0_client-js_1_tablet



I think what is happening is a mix of puppeteer's `captureBeyondScreenshot` option being `true` [1] (the default in backstop and puppeteer) which leads to a resize of the window before the screenshot and echo/ooui not handling the resize correctly.


[1] https://pptr.dev/#?product=Puppeteer&version=v14.1.2&show=api-pagescreenshotoptions

File: src/seedDb.sh [hunk 2]
```diff
@@ -1,4 +1,4 @@
 #!/usr/bin/env bash
 
-curl "https://pixeltool.netlify.app/database_2022-05-05_14-29-49-0600(MDT).tar.gz" -o var/lib/mysql/database.tar.gz
+curl "https://nicholasray.github.io/pixel-seed-data/database_2022-05-26_13-03-37-0700(PDT).tar.gz" -o var/lib/mysql/database.tar.gz
```
Comment (nicholasray) [inline src/seedDb.sh:3]: Note, I updated this to point to your seed data

File: src/engine-scripts/puppet/echo.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+			hidden: true
+		} );
+		// wait for a notification to appear if we're expecting them.
+		if ( expectsNotifications ) {
+			await page.waitForSelector( '.mw-echo-ui-notificationItemWidget-content, ' );
+		} else {
+			// Check for the "There are no notifications" message.
+			// Annoyingly there is no unique selector we can use, so we must use a delay
+			// to avoid false positives.
+			await delay( 5000 );
```
[the remark this answers is not in the document]
Comment (jdlrobson) [reply src/engine-scripts/puppet/echo.js:51]: (No see comment) I think this task might be the problem: https://phabricator.wikimedia.org/T287132

**Comment by jdlrobson** [comment]
> For some reason, the `SCENARIO > Talk:Test (#vector)` scenario is hanging (and never completing). You should be able to replicate it with `./pixel.js reference` (please let me know if you can't)

Yep I can replicate. Debugging as we speak..

File: configEcho.js (hunk 1 above)
Comment (jdlrobson) [reply configEcho.js]: Fixed in d94d0eabab34969

**Comment by jdlrobson** [comment]
Okay this should be ready for another review @nray !

**Pull request merged**